### PR TITLE
feat(LifeC): Make `doDamage` return whether or not health reached 0

### PR DESCRIPTION
**Pull request by Dankgard**

### CODE REVIEW

File: src/LifeC.cpp [hunk 1]
```diff
@@ -24,13 +24,14 @@ int LifeC::getTotalLife() { return totalLife; }
...
+bool LifeC::doDamage(float _damage) {
     if (!invulnerability)
         currentLife -= _damage;
 
     if (currentLife <= 0) {
-        scene->deleteEntity(father);
-    }
+        return true;
+    } else
+        return false;
```
Comment (kyranet) [inline src/LifeC.cpp:34]:
```cpp
return currentLife <= 0;
```

Comment (kyranet) [inline src/LifeC.cpp]:
```cpp
return currentLife <= 0;
```

File: src/LifeC.h [hunk 2]
```diff
@@ -34,8 +34,8 @@ class LifeC : public Component {
     // set the total life
     void setTotalLife(int _life);
 
-    // subtract the damage to the life
-    void doDamage(float _damage);
+    // subtract the damage to the life (true if currentLife reaches 0)
```
Comment (kyranet) [inline src/LifeC.h]:
```suggestion
    // Subtract the damage to the life. Returns a `boolean` value defining
    // whether or not `currentLife` reached 0.
```

File: src/MeleeEnemyBehaviourEC.cpp [hunk 3]
```diff
@@ -26,7 +26,10 @@ void MeleeEnemyBehaviourEC::checkEvent() {
                 // attack player
                 LifeC* playerHealth = dynamic_cast<LifeC*>(
                     scene->getEntitybyId("Player")->getComponent("LifeC"));
-                playerHealth->doDamage(getAttack());
+                // if player dies sleep method is called
+                if (playerHealth->doDamage(getAttack()))
+                    ;
+                // TO DO: call sleep()
```
Comment (kyranet) [inline src/MeleeEnemyBehaviourEC.cpp]:
```suggestion
                // TODO(MiriamLeis): call `sleep()` when funcionality is available
```

File: src/BulletHurtPlayerEC.cpp [hunk 4]
```diff
@@ -25,7 +25,10 @@ void BulletHurtPlayerEC::checkEvent() {
         int damage = bullet->getDamage();
         LifeC* playerHealth = dynamic_cast<LifeC*>(
             scene->getEntitybyId("Player")->getComponent("LifeC"));
-        // playerHealth->doDamage(damage);
+        /* // if player dies sleep method is called
+        if (playerHealth->doDamage(damage))
+            ;
+        // TO DO: call sleep() */
```
Comment (kyranet) [inline src/BulletHurtPlayerEC.cpp]: I'm pretty sure you want to call doDamage, and that you wanted to comment was the sleep not being available yet.
```suggestion
        // if player dies sleep method is called
        if (playerHealth->doDamage(damage))
            ;
        // TODO(MiriamLeis): call `sleep()` when funcionality is available
```

**Review by kyranet — CHANGES_REQUESTED** [review]

File: src/BulletHurtPlayerEC.cpp [hunk 5]
```diff
@@ -25,7 +25,11 @@ void BulletHurtPlayerEC::checkEvent() {
         int damage = bullet->getDamage();
         LifeC* playerHealth = dynamic_cast<LifeC*>(
             scene->getEntitybyId("Player")->getComponent("LifeC"));
-        // playerHealth->doDamage(damage);
+
+        // if player dies sleep method is called
+        if (playerHealth->doDamage(damage))
+            ;
+        // TO DO: call sleep()
```
Comment (kyranet) [inline src/BulletHurtPlayerEC.cpp]:
```suggestion
        // TODO(MiriamLeis): call `sleep()` when funcionality is available
```

**Review by kyranet — CHANGES_REQUESTED** [review]

**Review by kyranet — APPROVED** [review]

**Review by pablojor — APPROVED** [review]

**Pull request merged**